src/tsconfig.json

```diff
@@ -0,0 +1,32 @@
+{
```

Comment:
Should this file be under "src" folder or "src/frontend" folder? It seems to me that backend will not use TypeScript, right?

Reply:
I'm hoping to use typescript everywhere i can. I deeply value that structure it adds.

Reply:
If typescript is used only in front-end, please move the "tsconfi.json" to be under "src/frontend/".

Reply:
Wait, if you're wanting to use ts for both ends (which, to be clear, I'm totally okay with), are we still going to go with just plain node and express as the frameworks instead? After a quick bit of research, MongoDB and express would be the best fit for our project as it stands (and set up the ts definitions from a repo like [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped) , but there ARE typescript-focused frameworks ( [NestJS](https://docs.nestjs.com/) , and [tsoa](https://github.com/lukeautry/tsoa) are some examples) that we could use instead if we're wanting to go that route.

Reply:
Please use nodejs for the backend, instead of netjs or tsoa in the course project of CS372. You can explore these typescript-focused frameworks in other projects.

Reply:
@zeshengchen NestJS, express, tsoa are all frameworks for Nodejs. Are you saying we can't use any framework(s) for the backend?

Reply:
Please use express, which will be discussed in the class soon.

Reply:
To be clear, I plan on using TypeScript for BOTH back & front.

I already know TS works with React as I've used it previously, and I have had no issues with Typescript recognizing Express or Mongoose insofar.

The TSConfig file is has one configuration issue that I've just noticed, however it has successfully been helping my IDE see and compile TS and TSX into JS on all accounts. I see no reason to remove it from our backend right now.

Reply:
Hi Zack, you are already familiar with Typescript. In this project, please try out nodejs and express, which we will cover in the class. It is always good to learn something new!

Reply:
BTW, on the first week when we talked about the course project, I have provided the basic requirements for the course project. Specifically, it requires that backend will be implemented based on nodejs/express. You can find the information from the slides on the course project.

Reply:
Express and NodeJS are frameworks for Javascript. Typescript simply adds type strictness and compiles into said Javascript. I will still be directly using Express and Node. 

Removing TS would only make coding this project less convenient. It has no effect on Express or Node whether I have it or not.

Reply:
Hi Zack, I understand your point. But in this project, please use Nodejs/Express for the backend, which is the same requirement for all course projects in this course and the requirement from the instructor. My view is that it is good for students to try out nodejs before moving to using Typescript-based framework, although it may be less convenient. Learning a "less" convenient framework is better to understand the basic components and functionalities from a low-layer view. If you still want to discuss, please feel free to stop by my office hours.

Reply:
We are not using a typescript based framework. We are still using express. We are just using typescript with express.

Literally all that is being added is type strictness. We are NOT using a different framework. Typescript itself is *not* a framework.

I have no intention in changing from express to a different framework. I, however, plan on just using typescript for type strictness while using expressjs.

I do not understand where the confusion here is. I will happily talk to you about this after class Monday or during office hours.

Reply:
If you will use Typescript only for static typing, that is fine for me that your team can apply typescript as the backend. If more than static typing will be used from Typescript, please use nodejs for backend. 

But before making the final decision on this, please talk to your group members on this. It will take some time to learn Typescript efficiently and set it up. Moreover, we will do not have time to discuss about Typescript in the class. Instead, we will focus on Nodejs and express. Here is a summary article comparing Typescript with Nodejs for backend: https://graffersid.com/typescript-vs-nodejs/ .

If a group member does not feel comfortable to use Typescript as the backend, I would suggest to use nodejs. Based on what I found through Google search, a recommendation is to learn nodejs first and then move to Typescript.  

Please let me know your group's final decision on nodejs or Typescript with only static typing.

Reply:
Have your group made the decision on nodejs or Typescript with only static typing?